client/src/storage/device.rs

```diff
@@ -217,24 +603,64 @@ impl SessionStore for DeviceSessionStore {
         &self,
         address: &ProtocolAddress,
     ) -> Result<Option<SessionRecord>, SignalProtocolError> {
-        todo!()
+        let addr = format!("{}", address);
+
+        match sqlx::query!(
+            r#"
+            SELECT
+                session_record
+            FROM
+                DeviceSessionStore
+            WHERE
+                address = ?
+            "#,
+            addr
+        )
+        .fetch_one(&self.pool)
+        .await
+        {
+            Ok(row) => SessionRecord::deserialize(
+                BASE64_STANDARD
+                    .decode(row.session_record)
+                    .map_err(|err| SignalProtocolError::InvalidArgument(format!("{}", err)))?
+                    .as_slice(),
+            )
+            .map(|res| Some(res)),
+            Err(_) => Ok(None),
```

Comment:
Are we sure that all errors can just boil down to None?

Reply:
Det gør de